src/GenAPI/Microsoft.DotNet.GenAPI/CSharpFileBuilder.cs

```diff
@@ -74,8 +76,9 @@ private void Visit(IAssemblySymbol assembly)
 
             Document document = project.AddDocument(assembly.Name, compilationUnit);
             document = Simplifier.ReduceAsync(document).Result;
-            compilationUnit = document.GetSyntaxRootAsync().Result!;
+            document = Formatter.FormatAsync(document, DefineFormattingOptions()).Result;
```

Comment:
Isn't there a synchronous API that you could call instead? Calling onto Result of the async call is an anti-pattern afaik.

Reply:
There is a synchronous API for `Formatter` class, but there are not for `Simplifier` class. Plus, synchronous API works with `SyntaxNode` directly and asynchronous - with Document. In order to work with the `Formatter` and the `Simplifier` in the same way, I'd prefer to have:
```C#
Document document = project.AddDocument(assembly.Name, compilationUnit);
document = Simplifier.ReduceAsync(document).Result;
document = Formatter.FormatAsync(document, DefineFormattingOptions()).Result;
```
rather than
```C#
Document document = project.AddDocument(assembly.Name, compilationUnit);
document = Simplifier.ReduceAsync(document).Result;
compilationUnit = document.GetSyntaxRootAsync().Result!;
compilationUnit = Formatter.Format(compilationUnit, Formatter.Annotation, _adhocWorkspace, DefineFormattingOptions());
```